src/core/halide2isl.cc

```diff
@@ -376,8 +382,10 @@ ScheduleTreeAndDomain makeScheduleTreeHelper(
     }
 
     // Recursively descend.
+    outer.push_back(op->name);
```

Comment:
Let's please make this functional style to avoid push/pop.
I.e. just construct a new vector, append op->name to it and let it go out of scope.